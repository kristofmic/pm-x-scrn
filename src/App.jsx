import React from 'react';
import { connect } from 'react-redux';

import { setWhoIAm } from './actions';

function App(props) {
  const { setWhoIAm, whoAmI } = props;

  return (
    <div className="container">
      <h1>I am {whoAmI}!</h1>
      <button className="btn btn-primary" onClick={setWhoIAm}>
        Set who I am
      </button>
    </div>
  );
}

const withRedux = connect(
  state => state,
  { setWhoIAm }
);

export { App };
export default withRedux(App);
