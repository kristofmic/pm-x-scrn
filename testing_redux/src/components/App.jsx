import React from 'react';

import withRedux from './withRedux';

function App(props) {
  const { setWhoIAm, whoAmI } = props;

  return (
    <div className="container">
      <h1>I am {whoAmI}!</h1>
      <button className="btn btn-primary" onClick={setWhoIAm}>
        Set who I am
      </button>
    </div>
  );
}

export { App };
export default withRedux(App);
