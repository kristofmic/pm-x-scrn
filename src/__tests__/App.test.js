import React from 'react';
import { shallow } from 'enzyme';

import { App } from '../App';

describe('<App/>', () => {
  const mockSetWhoIAm = jest.fn();

  beforeEach(() => {
    mockSetWhoIAm.mockClear();
  });

  test('should match the snapshot', () => {
    const app = shallow(<App whoAmI="Thanos" setWhoIAm={mockSetWhoIAm} />);
    expect(app.html()).toMatchSnapshot();
  });

  test('should call setWhoIAm on click', () => {
    const app = shallow(<App whoAmI="Thanos" setWhoIAm={mockSetWhoIAm} />);
    expect(mockSetWhoIAm).toHaveBeenCalledTimes(0);
    app.find('button').simulate('click');
    expect(mockSetWhoIAm).toHaveBeenCalledTimes(1);
  });
});
