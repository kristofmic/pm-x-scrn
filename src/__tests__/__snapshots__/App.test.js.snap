// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should match the snapshot 1`] = `"<div class=\\"container\\"><h1>I am Thanos!</h1><button class=\\"btn btn-primary\\">Set who I am</button></div>"`;
